tinker/static/assets/js/tinker.js

```diff
@@ -14,4 +14,117 @@ $(document).ready(function () {
             $(this).text("- Less Search Filters")
         }
     });
-});
\ No newline at end of file
+});
+
+function pagination() {
+    var numberOfItems = $(" #loop .list-group").length;
+    var limitPerPage = $("#first").val();
+
+    $("#selected-option").change(function() {
+        var o = $(this).find("option:selected");
+        if (o.val() === "All") {
+            limitPerPage = numberOfItems;
+        } else {
+            limitPerPage = o.val()
+        }
+
+        $(".pagination").children("li.to-remove").remove();
+
+        var totalPages = Math.ceil(numberOfItems / limitPerPage);
+
+        createButtons(totalPages, limitPerPage);
+
+        var currentPage = 1;
+
+        var grandTotal = limitPerPage * currentPage;
+
+        for (var i = grandTotal - limitPerPage; i < grandTotal ; i ++) {
+            $("#loop .list-group:eq(" + i + ")").show();
+        }
+
+        switchPages(limitPerPage);
+
+        nextPage(totalPages, limitPerPage);
+
+    });
+
+    var totalPages = Math.ceil(numberOfItems / limitPerPage);
+
+    createButtons(totalPages, limitPerPage);
+
+    switchPages(limitPerPage);
+
+    nextPage(totalPages, limitPerPage);
+
+    previousPage(totalPages, limitPerPage);
+}
+
+function nextPage(totalPages, limitPerPage) {
+    $("#next-page").on("click", function() {
+        var currentPage = $(".pagination li.active").index();
+        if (currentPage === totalPages) {
+            return false;
+        } else {
+            currentPage ++;
+            $(".pagination li").removeClass("active");
+            $("#loop .list-group").hide();
+
+            var grandTotal = limitPerPage * currentPage;
+
+            for (var i = grandTotal - limitPerPage; i < grandTotal ; i ++) {
+                $("#loop .list-group:eq(" + i + ")").show();
+            }
+            $(".pagination li.current-page:eq(" + (currentPage - 1) + ")").addClass("active");
+        }
+    });
+}
+
+function previousPage(totalPages, limitPerPage) {
+    $("#previous-page").on("click", function() {
```

Comment:
I feel like both previousPage and nextPage should only call a single method and pass parameters (what page # you want to be one and how many to show).

Reply:
Combined nextPage and previousPage to become a single method